### ISSUE TITLE
Add features for the Statemachine example

### ISSUE THREAD

**Issue by pluralia** [eclipse-langium/langium#1257]
* validation
  - [ ] Validate unreachable states
  - [ ] Validate unused events

* code actions
  - [ ] If a state name doesn't start with a capital letter, replace it with capitalized name (look at the corresponding validation [here](https://github.com/eclipse-langium/langium/blob/main/examples/statemachine/src/language-server/statemachine-validator.ts#L28-L35))
  - [ ] Remove an unreachable state
  - [ ] Remove an unused event

* LSP
   - [ ] `Formatter`
   - [ ] `RenameProvider`

**Comment by Yokozuna59** [comment]
Hi @pluralia,

I plan to work on this if it's possible, but I've few questions:

1. On the first code actions:

    > If a state name doesn't start with a capital letter, replace it with capitalized name (look at the corresponding validation [here](https://github.com/eclipse-langium/langium/blob/main/examples/statemachine/src/language-server/statemachine-validator.ts#L28-L35))
  
    The action should also update everywhere the state has been used, shouldn't it?

2. Do we actually need to override `RenameProvider`? It's working fine:
  
    [Screencast from 2023-11-04 15-02-12.webm](https://github.com/eclipse-langium/langium/assets/87622592/2c5cdf07-4890-49da-8dc0-3da0459db7af)

3. Should we also validate commands uniqueness and usage? Currently, it's possible to have duplicate command names within commands, states, and events.

**Comment by msujew** [comment]
@Yokozuna59 

> The action should also update everywhere the state has been used, shouldn't it?

Yes, it should work similar to the rename LSP feature in that regard.

> Do we actually need to override RenameProvider? It's working fine:

It was probably added in error, I don't think it's necessary.

> Should we also validate commands uniqueness and usage? Currently, it's possible to have duplicate command names within commands, states, and events.

Yep, that would be good 👍

**Comment by Yokozuna59** [comment]
@msujew Thanks for the feedback. I'm almost done, but the formatting is kind of performing some of the expected behaviors. For example, the following format code:

```ts
formatter.keyword('initialState')
    .prepend(Formatting.fit(Formatting.newLine({ allowMore: true }), /* another rule */))
    .append(Formatting.oneSpace());
```

It formats the previous lines and makes them one, although it has `allowMore`. If I remove the `Formatting.fit`, it doesn't do that.

I tried to add and remove another rule in `Formatting.fit`, but it's performing the same behavior.
